[remuslum] iP

remuslum opened the pull request:
**Duke ChatBot** 
> Your mind is for having ideas, not holding them." - David Allen

[How a chatbot works](https://www.google.com)

Duke brings convenience into your life by:

- Having a to-do list
- Marking your tasks as completed/uncompleted
- Add/Delete tasks at will

All you need to do is:

1. Input your task as todo/deadline/event
2. Mark/Unmark the tasks as completed 👍🏻 

Things to take note:

- [ ] Have you inputted the task in the correct format?
- [ ] Understand the error message that is shown to you and make the corresponding changes?

Here is the main method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

remuslum committed:
Added Rename, Greet and Exit

remuslum committed:
Added Echo

remuslum committed:
Added Add, List

remuslum committed:
Added Done and Undone

remuslum committed:
Added ToDos, Events, Deadlines

remuslum committed:
Added Automated Text UI testing

remuslum committed:
Added Error Handling

remuslum committed:
Added Delete

remuslum committed:
Added enum class for error messages

remuslum committed:
Added enum class for error messages

remuslum committed:
Revert "Added enum class for error messages"

This reverts commit a8a6878a9ac6a40baacc6f8cdc5576be9cd84af3.

remuslum committed:
Modified the input and Expected textfiles

remuslum committed:
Added duke.txt and save/load methods

remuslum committed:
Added DateTime Parser

remuslum committed:
Added all necessary Classes

remuslum committed:
Merge pull request #1 from remuslum/A-MoreOOP

Added all necessary Classes

remuslum committed:
Added Packages for all classes

remuslum committed:
Merge pull request #2 from remuslum/packages

Added Packages for all classes

remuslum committed:
Merge branch 'master' into add-gradle-support

remuslum committed:
Added JUnit Test Classes

remuslum committed:
Added JAR file

remuslum committed:
Added JavaDoc for all classes

remuslum committed:
Applied Coding Standard

remuslum committed:
Added FindCommand

remuslum committed:
Added FindCommand

remuslum committed:
Merge branch 'branch-level-9'

yuxunn commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,116 @@
+            }
+        }
+        else {
+            throw new DukeException(ErrorMessages.INCOMPREHENSIBLE_TASK.getMessage());
```
Thats a lot of if-else statements 😮 perhaps you could find a way to reduce the number of if-else statements to increase code readability.

yuxunn commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,116 @@
+import duke.command.*;
+import duke.messages.ErrorMessages;
+import duke.messages.Messages;
+import duke.task.*;
```
imported classes should be listed explicitly :)

PearlynnT replied:
I agree with this comment.

yuxunn commented on src/test/java/EventTest.java:
```diff
@@ -0,0 +1,19 @@
+        LocalDateTime endDate = LocalDateTime.parse("2023-09-01 1PM", formatter);
+        String taskName = "project meeting";
+
+        assertEquals("[E] [ ] project meeting (from: Aug 31 2023 2PM to: Sep 1 2023 1PM)", new Event(taskName, startDate, endDate).toString());
```
i think you can place line break here to improve readability 😄

yuxunn commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+            } else if (task.contains("D")){
+                String[] deadlineString = task.substring(8).split("\\(");
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy ha");
+                LocalDateTime deadlineDate = LocalDateTime.parse(deadlineString[1].substring(4, deadlineString[1].length()-1), formatter);
```
line break here

yuxunn reviewed (commented):
overall i think your code looks rly good, just some really minor changes! good job!! 😃

remuslum committed:
Added CheckStyle and Set Up GUI

PearlynnT commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,186 @@
+/**
+ * Main class of the application.
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,116 @@
+/**
+ * Parses the user input into the respective command.
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,59 @@
+/**
+ * Handles the textfile provided by the user and loads all tasks from previous sessions.
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,59 @@
+     * @return an ArrayList of the existing tasks in String format
+     * @throws DukeException if the filepath is incorrectly inputted
+     */
+
```
There should be no blank line between the documentation block and the method.

PearlynnT commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+/**
+ * Handles all operations of the taskList.
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+package duke;
+
+import duke.task.*;
```
Imported classes should always be listed explicitly.

PearlynnT commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Shows all appropriate messages to the user.
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,45 @@
+/**
+ * A command that lists all task associated with the keyword inputted by the user
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,22 @@
+/**
+ * Display all tasks in the current task list
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,22 @@
+     * @param taskList the existing task list of the user
+     * @param ui the ui that handles successful/unsuccessful messages
+     */
+
```
There should be no blank line between the documentation block and the method.

PearlynnT commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,41 @@
+/**
+ * A command that marks a specified task as Completed
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,41 @@
+     * @param ui the ui that handles successful/unsuccessful messages
+     * @throws DukeException If the task has already been marked, an error is thrown
+     */
+    public void execute(TaskList taskList, Ui ui)throws DukeException {
```
```suggestion
    public void execute(TaskList taskList, Ui ui) throws DukeException {
```

PearlynnT commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,41 @@
+     */
+    public void execute(TaskList taskList, Ui ui)throws DukeException {
+        try {
+            Task taskToBeMarked= taskList.getTask(this.taskNumber);
```
```suggestion
            Task taskToBeMarked = taskList.getTask(this.taskNumber);
```

PearlynnT commented on src/main/java/duke/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * A command that unmarks a specified task
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,40 @@
+     * @param ui the ui that handles successful/unsuccessful messages
+     * @throws DukeException If the task has already been unmarked, an error is thrown
+     */
+
```
There should be no blank line between the documentation block and the method.

PearlynnT commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,26 @@
+/**
+ * Represents an Event Task.
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,39 @@
+/**
+ * Represents a task to be added by the user.
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Represents a Todo task.
+ */
+
```
There should be no blank line between the documentation block and the class.

PearlynnT commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,59 @@
+package duke;
+
+import java.io.*;
```
Imported classes should always be listed explicitly.

PearlynnT commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,39 @@
+
+public class Task {
+    protected String name ;
+    protected boolean status ;
```
```suggestion
    protected boolean isCompleted;
```
Boolean variables/methods should be named to sound like booleans

PearlynnT commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,18 @@
+     * @param taskList the existing task list of the user.
+     * @param ui the ui that handles successful/unsuccessful messages
+     */
+    public void execute(TaskList taskList, Ui ui){
```
Unused parameters in method signature.

PearlynnT commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+        this.taskList = new ArrayList<>();
+        for (String task: tasks){
+            if (task.contains("T")){
+                Todo convertedToDoTask = new Todo(task.substring(8));
```
Why is the todo named as convertedToDoTask?

PearlynnT reviewed (commented):
Hi you can consider editing your code to follow the coding standard. Your code is generally readable, good job so far :smile:

remuslum committed:
Modified GUI Code to launch Duke ChatBot

remuslum committed:
Included JUnit Test Cases for add, delete and mark

remuslum committed:
Modified code to fit style

remuslum committed:
Modified the subtraction of index to the individual commands

remuslum committed:
Added UnmarkCommand JUnit test

remuslum committed:
Merge branch 'branch-jUnit-2'

remuslum committed:
Created and enabled assert statements

remuslum committed:
Merge pull request #3 from remuslum/branch-assert

Created and enabled assert statements

remuslum committed:
Added Help Command and test cases for it

remuslum committed:
Merge pull request #4 from remuslum/branch-C-Help

Added Help Command and test cases for it

remuslum committed:
Modified code to make main path obvious

remuslum committed:
Code currently violates several coding quality standards
This affects the readability of the code and cause possible confusion.
As such, thw code has been modified such that class names, variables names all provide meanings and if-else statements are modified such that the main body is obvious to the reader.

remuslum committed:
Added and tweaked GUI components

remuslum committed:
Added Test cases for Find and List Commands

remuslum committed:
Modified Code to fit checkstyle

remuslum committed:
Added product screenshots

remuslum committed:
Added User Guide

remuslum committed:
Finalised features and JAR file created

remuslum committed:
Updated README.md

remuslum committed:
Modified constructor to pass smoke tests

remuslum committed:
Merge branch 'master' of https://github.com/remuslum/ip
